server.js

```diff
@@ -34,86 +34,125 @@ app.use(bodyParser.json(), urlencodedParser);
 app.use(cors());
 
 app.listen(process.env.PORT, () => {
-  console.log(`App is Listening on PORT ${process.env.PORT}`)
+  console.log(`App is Listening on PORT ${process.env.PORT}`);
 });
 
 app.get("/", async (req, res) => {
   try {
     const users = await userModel.find();
-    console.log(users)
-    res.json({ message: "Success" })
-  }
-  catch (error) {
-    res.status(503).json({ msg: 'Cant reach server' })
+    console.log(users);
+    res.json({ message: "Success" });
+  } catch (error) {
+    res.status(503).json({ msg: "Cant reach server" });
   }
 });
 
 app.get("/user", async (req, res) => {
   try {
-    const user = await userModel.findOne({ email: req.query.email })
-    const password = req.query.password
-    if (!user) {
-      res.status(404).json({ msg: 'User not found.' })
-    }
-    else if (password !== user.password) {
-      res.status(403).json({ msg: 'Incorrect password.' })
-    }
-    else {
-      res.status(200).json({ userUID: user._id, name: user.username, email: user.email })
+    const user = await userModel.findOne({ email: req.query.email });
```

Comment:
MongoDB is queried here before any data validation has been done.  If neither a valid e-mail address nor a password has been provided then there's no point in querying MongoDB -- we already know that we're going to send an error response to the frontend.

Please move this call to inside a new code block starting after the `else` on line 57 so that it looks something like this:

```javascript
    } else {
      try {
        const user = await userModel.findOne({ email });
      } catch (error) {
        res.status(503).json({ msg: "Cant reach server" });
      }
      if (!user) {
        res.status(404).json({ msg: "User not found." });
```